apps/authentication.py

```diff
@@ -36,15 +36,16 @@ class AsyncJsonAuthConsumer(AsyncJsonWebsocketConsumer):
         await super().disconnect(code)
 
     async def receive(self, text_data=None, bytes_data=None, **kwargs):
-        if await self._authenticate():
-            if text_data:
-                json = await self.decode_json(text_data)
-                if 'event' in json and 'data' in json and json['event'] == 'refresh_jwt':
-                    self.scope['jwt'] = json['data']
-                else:
-                    await self.receive_json(json, **kwargs)
-            else:
-                raise ValueError("No text section for incoming WebSocket frame!")
+        if text_data:
+            json = await self.decode_json(text_data)
+            # Handle jwt_refresh requests before attempting authentication
+            refresh_request = 'event' in json and 'data' in json and json['event'] == 'refresh_jwt'
+            if refresh_request:
+                self.scope['jwt'] = json['data']
+            if await self._authenticate() and not refresh_request:
```

Comment:
Wouldn't it make more sense to just have this as an `elif await self._authenticate()` as the else part would handle the `not refresh_request` part of the if statement

Reply:
No, we want `self._authenticate()` to run with the new JWT to ensure that the refreshed JWT is valid (and close the socket if the refreshed JWT is invalid).